File: src/main/java/com/aethercraft/moba/util/MOBAExecutor.java
package com.aethercraft.moba.util;

import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;

import com.aethercraft.moba.MOBACraft;

public class MOBAExecutor implements CommandExecutor {
	
	private MOBACraft plugin;
	
	public MOBACraft getPlugin() {
		return plugin;
	}

	public void setPlugin(MOBACraft plugin) {
		this.plugin = plugin;
	}
	
	public MOBAExecutor (MOBACraft plugin) {
		this.setPlugin(plugin);
	}

	public boolean onCommand(CommandSender sender, Command cmd, String label, String[] args) {
		if (args.length > 4) {
			sender.sendMessage("Too many arguments.  Both sender and cmd are required.");
			return false;
		}
		if (args.length < 2) {
			sender.sendMessage("Not enough arguments.  Both sender and cmd are required.");
			return false;
		}
		// implementation
		return false;
	}

}

File: src/main/scala/com/aethercraft/moba/util/MOBAExecutor.scala
package com.aethercraft.moba.util

import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import com.aethercraft.moba.MOBACraft
import scala.beans.BeanProperty

class MOBAExecutor(@BeanProperty val plugin: MOBACraft) extends CommandExecutor {

  def onCommand(sender: CommandSender, cmd: Command, label: String, args: Array[String]) = {
    args match {
      case args if args.length > 4 =>
        sender.sendMessage("Too many arguments.  Both sender and cmd are required.")
        false
      case args if args.length < 2 =>
        sender.sendMessage("Not enough arguments.  Both sender and cmd are required.")
        false
      case _ =>
        // implementation
        true
    }
  }
}